### ISSUE TITLE
valid-jsdoc.preferType not working with multiple types on optional parameter

### ISSUE THREAD

**Issue by patric-eberle** [eslint/eslint#10455]
**Tell us about your environment**
macOS 10.13.4

* 4.19.1
* 10.3.0
* 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
var capitalizedCommentsIgnore = [
  // Vue
  'activated\\(\\)',
  'components:',
  'computed:',
  'created\\(\\)',
  'deactivated\\(\\)',
  'destroyed\\(\\)',
  'vue',
  'watch:',

  // Ember
  'ember'
].join('|');

module.exports = {
  extends: [
    'eslint-config-airbnb-base',
  ],
  env: {
    browser: true,
    node: true,
    amd: false,
    mocha: false,
    jasmine: false
  },
  plugins: ['import'],
  settings: {
    'import/core-modules': [
      // Ember.js
      '../../resolver',
      '../../app',
      './config/environment',
      'ember',
      'ember-qunit',
      'htmlbars-inline-precompile',
      'qunit'
    ]
  }
  rules: {
    'import/extensions': [1, 'never'],
    'import/no-extraneous-dependencies': 0,
    'import/no-mutable-exports': 1,
    'import/no-unresolved': 0
  },
  globals: {},
  rules: {
    // Enforce return statements in callbacks of array methods
    // - Allowed to use .map() instead forEach without a return value
    'array-callback-return': 0,

    // Enforce consistent brace style for all control statements
    'curly': [2, 'all'],

    // Enforce consistent newlines before and after dots
    'dot-location': [2, 'property'],

    // Disallow reassignment of function parameters
    // Disallow parameter object manipulation
    // Rule: http://eslint.org/docs/rules/no-param-reassign.html
    // - Allowed to support default fallback: var foo = foo || 'baa';
    'no-param-reassign': 0,

    // Disallow unmodified conditions of loops
    // http://eslint.org/docs/rules/no-unmodified-loop-condition
    'no-unmodified-loop-condition': 2,

    // Disallow unnecessary .call() and .apply()
    'no-useless-call': 2,

    // Disallow usage of configurable warning terms in comments.
    'no-warning-comments': [1, {
      terms: ['fixme', 'xxx', 'console', 'stupid', 'dumb', 'fuck']
    }],
    // Require or disallow trailing commas
    // https://eslint.org/docs/rules/comma-dangle
    'comma-dangle': 0,

    // Disallow use of constant expressions in conditions
    'no-constant-condition': 2,

    // Disallow unnecessary parentheses
    // http://eslint.org/docs/rules/no-extra-parens
    'no-extra-parens': [2, 'all', {
      returnAssign: false, // Allows parenthesis in fat arrow returns
      nestedBinaryExpressions: false,
      enforceForArrowConditionals: false // Allows parenthesis for condition ins fat arrow returns
    }],

    // Avoid code that looks like two expressions but is actually one
    'no-unexpected-multiline': 2,

    // Ensure JSDoc comments are valid
    // http://eslint.org/docs/rules/valid-jsdoc
    // - force uniform jsdoc style
    'valid-jsdoc': [2, {
      requireReturnDescription: false,
      requireReturn: false,
      prefer: {
        return: 'returns',
        arg: 'param',
        argument: 'param'
      },
      preferType: {
        array: 'Array',
        boolean: 'Boolean',
        function: 'Function',
        node: 'Node',
        // 'null' is only valid in lower case
        number: 'Number',
        object: 'Object',
        promise: 'Promise',
        string: 'String'
      },
      matchDescription: '.+\\.($|\\n)'
    }],
    // require constructor function names to begin with a capital letter
    // - Adding improved support for ember
    'new-cap': [2, {
      newIsCap: true,
      capIsNewExceptions: ['A']
    }],

    // require or disallow method and property shorthand syntax for object literals
    'object-shorthand': [1, 'always', {
      avoidQuotes: true
    }],

    // enforce spacing between rest and spread operators and their expressions
    'rest-spread-spacing': [2, 'never'],

    // require const declarations for variables that are never reassigned after declared
    'prefer-const': 1,

    rules: {
      // Enforce or disallow capitalization of the first letter of a comment.
      'capitalized-comments': [1, 'always', {
        ignorePattern: '[' + capitalizedCommentsIgnore + ']',
        ignoreInlineComments: true // Allow commenting of function params and similar.
      }],

      // Enforces consistent naming when capturing the current execution context.
      'consistent-this': [2, '_that'],

      // Require or disallow named function expressions.
      'func-names': 0,

      // This option enforces minimum and maximum identifier lengths.
      // (variable names, property names etc.)
      'id-length': [1, {
        min: 3,
        properties: 'always',
        exceptions: [
          '$',
          'a',
          'b',
          'e',
          'i',
          'in', // ember-cp-validations
          'j',
          'x',
          'y',
          '_', // _Lowdash
          'fs', // ember
          'gt', // ember-cp-validations
          'id',
          'DS' // ember-data
        ]
      }],

      // Enforce the location of arrow function bodies with implicit returns.
      // https://eslint.org/docs/rules/implicit-arrow-linebreak
      'implicit-arrow-linebreak': [2, 'beside'],

      // Enforce consistent spacing between keys and values in object literal properties.
      // http://eslint.org/docs/rules/key-spacing
      // - Allow value alignment for multiline.
      'key-spacing': [2, {
        singleLine: {
          mode: 'strict'
        },
        multiLine: {
          mode: 'minimum'
        }
      }],

      // enforces empty lines around comments.
      // - Force cleaner comment style.
      'lines-around-comment': [2, {
        beforeBlockComment: true,
        allowBlockStart: true,
        allowObjectStart: true
      }],

      // Specify the maximum length of a line in your program.
      // http://eslint.org/docs/rules/max-len
      // - Weaken max line length rule.
      'max-len': [1, 130, 2, {
        ignoreTrailingComments: true,
        ignoreUrls: true
      }],

      // enforce a maximum number of parameters in function definitions.
      // - Force max number of function params.
      'max-params': [2, 3],

      // Disallow bitwise operators.
      'no-bitwise': 1,

      // Disallow use of the continue statement.
      'no-continue': 1,

      // Disallow if as the only statement in an else block.
      'no-lonely-if': 1,

      // Disallow mixes of different operators
      // - Force cleaner code style
      'no-mixed-operators': 2,

      // Disallow multiple empty lines and only one newline at the end.
      'no-multiple-empty-lines': [2, { max: 1, maxEOF: 1 }],

      // Disallow negated conditions.
      // http://eslint.org/docs/rules/no-negated-condition
      // - Encourage cleaner code style.
      'no-negated-condition': 0,

      // Disallow trailing whitespace at the end of lines.
      // - Weaken for better IDE support.
      'no-trailing-spaces': [1, {
        ignoreComments: true,
        skipBlankLines: true
      }],

      // Disallow dangling underscores in identifiers (no-underscore-dangle).
      // http://eslint.org/docs/rules/no-underscore-dangle
      // - Allow to touch private values of this.
      'no-underscore-dangle': [2, {
        allowAfterThis: true,
        allow: ['_that']
      }],

      // Enforce consistent line breaks inside braces.
      // - Currently (2016-06-23) not working with single properties.
      // 'object-curly-newline': [2, {
      //   'multiline': true,
      //   'minProperties': 2
      // }],

      // Enforce placing object properties on separate lines.
      // - Encourage cleaner code style.
      // - Number of attributes is handled by object-curly-newline.
      'object-property-newline': [1, {
        allowAllPropertiesOnSameLine: true
      }],

      // Require assignment operator shorthand where possible or prohibit it entirely.
      // - Encourage cleaner code style
      'operator-assignment': [1, 'always'],

      // Enforce operators to be placed before or after line breaks.
      // - Encourage cleaner code style.
      'operator-linebreak': [1, 'before'],

      // Require or disallow padding lines between statements.
      // http://eslint.org/docs/rules/padding-line-between-statements
      'padding-line-between-statements': [
        'error',
        {
          blankLine: 'always',
          prev: ['const', 'let', 'var'],
          next: '*'
        },
        {
          blankLine: 'any',
          prev: ['const', 'let', 'var'],
          next: ['const', 'let', 'var']
        },
        {
          blankLine: 'always',
          prev: '*',
          next: 'case'
        },
        {
          blankLine: 'always',
          prev: '*',
          next: 'default'
        },
        {
          blankLine: 'always',
          prev: 'directive',
          next: '*'
        },
        {
          blankLine: 'any',
          prev: 'directive',
          next: 'directive'
        },
        {
          blankLine: 'always',
          prev: '*',
          next: 'if'
        },
        {
          blankLine: 'always',
          prev: '*',
          next: 'return'
        }
      ],

      // Require quotes around object literal property names.
      // - Force cleaner code style
      'quote-props': [2, 'consistent-as-needed'],

      // Do not require jsDoc.
      // http://eslint.org/docs/rules/require-jsdoc
      // - Force code documentation
      'require-jsdoc': [2, {
        require: {
          FunctionDeclaration: true,
          MethodDefinition: true,
          ClassDeclaration: true,
          ArrowFunctionExpression: true,
          FunctionExpression: true
        }
      }],

      // Enforce consistent spacing before function definition opening parenthesis.
      'space-before-function-paren': [2, 'never'],

      // Require or disallow spaces before/after unary operators.
      // - Force cleaner code style
      'space-unary-ops': 2,

      // Require regex literals to be wrapped in parentheses.
      // - Force cleaner code style
      'wrap-regex': 2,
      // Disallow labels that share a name with a variable.
      // - Force better code style
      'no-label-var': 2,

      // Disallow use of undefined when initializing variables.
      // - Force better code style
      'no-undef-init': 2,

      // Disallow use of undefined variable.
      // - Force better code style
      'no-undefined': 2
    }
  }
};
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
/**
   * This demonstrates a bug in ESLint 4.19.1 in combination with valid-jsdoc.preferType.
   * 
   * @param {object} mandatory1 - This line gives an error about lowercase 'object'
   * @param {object|object[]} mandatory2 - This line gives two errors about lowercase 'object'
   * @param {object} [optional1] - This line gives an error about lowercase 'object'.
   * @param {object|object[]} [optional2] - This line gives NO ERROR about lowercase 'object'.
   * 
   * @returns {Object}
   */
  function optionalParams(mandatory1, mandatory2, optional1, optional2) {
    return {};
  }
```

```bash
eslint tests/** --config index.js
```

**What did you expect to happen?**
ESLint should throw an error for 

```js
/**
   * @param {object|object[]} [optional2] - This line gives NO ERROR about lowercase 'object'.
   */
```
because `object` should start if an upper case.

**What actually happened? Please include the actual, raw output from ESLint.**
No error is thrown. The square brackets that mark this parameter as optional seem to break the behaviour/test. It works for single types.

```bash
65:14  error    Use 'Object' instead of 'object'                                             valid-jsdoc
66:14  error    Use 'Object' instead of 'object'                                             valid-jsdoc
66:21  error    Use 'Object' instead of 'object'                                             valid-jsdoc
67:14  error    Use 'Object' instead of 'object'                                             valid-jsdoc
```

**Comment by nzakas** [comment]
Thanks for the suggestion. As of yesterday, we have decided to officially [end-of-life JSDoc support](https://eslint.org/blog/2018/11/jsdoc-end-of-life) in ESLint. All JSDoc features are now deprecated and we won't be fixing bugs or making any further improvements to those features.

We are recommending that people transition over to use the [`eslint-plugin-jsdoc`](https://github.com/gajus/eslint-plugin-jsdoc) plugin instead of the built-in rules in ESLint.

Thanks for understanding and we apologize for any inconvenience.
